packages/jest-mock/src/index.ts

```diff
@@ -3,22 +3,34 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 
-import type {Global} from 'types/Global';
-
-type Mock = any;
-export type MockFunctionMetadata = {
-  ref?: any,
-  members?: {[key: string]: MockFunctionMetadata},
-  mockImpl?: () => any,
-  name?: string,
-  refID?: string | number,
-  type?: string,
-  value?: any,
-  length?: number,
+type Global = NodeJS.Global;
+
+type MockFunctionMetadataType =
+  | 'object'
+  | 'array'
+  | 'regexp'
+  | 'function'
+  | 'constant'
+  | 'collection'
+  | 'null'
+  | 'undefined';
+
+// TODO: bring this type export back once done with TS migration
+type MockFunctionMetadata<
```

Comment:
You can put this in `@jest/types` and import it here (if we need to be able to import this type anywhere else). `ModuleMocker` is just `typeof import('jest-diff')` so that should be fine

Reply:
done 06286d2